src/Test.UnitTests.Sarif/Core/SerializedPropertyInfoTests.cs

```diff
@@ -0,0 +1,171 @@
+                    ExpectedCompareResult = -1
+                },
+                new
+                {
+                    Left = new SerializedPropertyInfo(JsonConvert.SerializeObject(testGuid), isString:true),
+                    Right = new SerializedPropertyInfo(JsonConvert.SerializeObject(testGuid), isString:true),
+                    ExpectedEqualsResult = true,
+                    ExpectedCompareResult = 0
+                },
+                new
+                {
+                    Left = new SerializedPropertyInfo(JsonConvert.SerializeObject(testGuid), isString:true),
+                    Right = new SerializedPropertyInfo(JsonConvert.SerializeObject(Guid.Empty), isString:true),
+                    ExpectedEqualsResult = false,
+                    ExpectedCompareResult = 1
+                },
+                new
+                {
+                    Left = new SerializedPropertyInfo(JsonConvert.SerializeObject(now), isString:true),
+                    Right = new SerializedPropertyInfo(JsonConvert.SerializeObject(now), isString:true),
+                    ExpectedEqualsResult = true,
+                    ExpectedCompareResult = 0
+                },
+                new
+                {
+                    Left = new SerializedPropertyInfo(JsonConvert.SerializeObject(now), isString:true),
+                    Right = new SerializedPropertyInfo(JsonConvert.SerializeObject(default(DateTime)), isString:true),
+                    ExpectedEqualsResult = false,
+                    ExpectedCompareResult = 1
+                },
+                new
+                {
+                    // "99" compares "\"99\""
+                    Left = new SerializedPropertyInfo(JsonConvert.SerializeObject(99), isString:false),
+                    Right = new SerializedPropertyInfo(JsonConvert.SerializeObject("99"), isString:true),
+                    ExpectedEqualsResult = false,
+                    ExpectedCompareResult = 1
+                },
+                new
+                {
+                    // serialzed string: "[\"a\",\"b\",\"c\"]"
+                    Left = new SerializedPropertyInfo(JsonConvert.SerializeObject(new[] { "a", "b", "c" }), isString:false),
+                    Right = new SerializedPropertyInfo(JsonConvert.SerializeObject(new[] { "a", "b", "c" }), isString:false),
+                    ExpectedEqualsResult = true,
+                    ExpectedCompareResult = 0
+                },
+                new
+                {
+                    Left = new SerializedPropertyInfo(JsonConvert.SerializeObject(new[] { "a", "b", "c" }), isString:false),
+                    Right = new SerializedPropertyInfo(JsonConvert.SerializeObject(new[] { "c", "b", "a" }), isString:false),
+                    ExpectedEqualsResult = false,
+                    ExpectedCompareResult = -1
+                },
+                new
+                {
+                    // serialzed string: "{\"a\":1,\"b\":true}"
```

Comment:
>// serialzed string: "{\"a\":1,\"b\":true}" [](http://example.com/codeflow?start=20&length=43)

by "addtional info that is not in the code" I think these comments can be removed